ResidentInformationApi/V1/Factories/ResponseFactory.cs

```diff
@@ -1,55 +1,197 @@
-                PhoneNumber = domain.PhoneNumberList?.ToResponse()
+                PhoneNumber = domain.PhoneNumberList?.ToResponse(),
+                EmailAddressList = domain.EmailAddressList?.ToResponse()
             };
         }
+
         public static List<HousingResidentInformationResponse> ToResponse(this IEnumerable<HousingResidentInformation> residents)
         {
             return residents.Select(r => r.ToResponse()).ToList();
         }
 
+        public static MosaicResidentInformationResultResponse ToResponse(this MosaicResidentInformationResult domain)
+        {
+            return new MosaicResidentInformationResultResponse
+            {
+                System = domain.System,
+                SystemId = domain.SystemId,
+                SystemUrl = domain.SystemUrl,
+                ResidentInformation = domain.ResidentInformation.ToResponse()
+            };
+        }
+
+        public static MosaicResidentInformationResponse ToResponse(this MosaicResidentInformation domain)
+        {
+            return new MosaicResidentInformationResponse
+            {
+                MosaicId = domain.MosaicId,
+                FirstName = domain.FirstName,
+                LastName = domain.LastName,
+                Uprn = domain.Uprn,
+                DateOfBirth = domain.DateOfBirth,
+                PhoneNumber = domain.PhoneNumberList.ToResponse(),
+                AddressList = domain.AddressList.ToResponse(),
+                NhsNumber = domain.NhsNumber
+            };
+        }
+
+        public static List<MosaicResidentInformationResponse> ToResponse(this List<MosaicResidentInformation> residents)
+        {
+            return residents.Select(r => r.ToResponse()).ToList();
+        }
         private static List<Phone> ToResponse(this List<PhoneNumber> phoneNumbers)
         {
-            return phoneNumbers.Select(number => new Phone
+            return phoneNumbers.Select(r => r.ToResponse()).ToList();
+        }
+
+        private static Phone ToResponse(this PhoneNumber domain)
+        {
+            return new Phone
+            {
+                PhoneNumber = domain.Number,
+                PhoneType = domain.Type.ToResponse()
+            };
+        }
+
+        private static PhoneTypeResponse ToResponse(this PhoneType domain)
+        {
+            switch (domain)
             {
-                PhoneNumber = number.Number,
-                PhoneType = number.Type
-            }).ToList();
+                case PhoneType.Primary:
+                    return PhoneTypeResponse.Primary;
+                case PhoneType.Home:
+                    return PhoneTypeResponse.Home;
+                case PhoneType.Mobile:
+                    return PhoneTypeResponse.Mobile;
+                case PhoneType.Fax:
+                    return PhoneTypeResponse.Fax;
+                default:
+                    throw new ArgumentException("Source type is not enum");
```

Comment:
This error could be more descriptive, maybe something like "Phone type is invalid"

Reply:
I'll amend it!